tests/kernel/pipe/pipe_api/src/test_pipe_avail.c

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2020 Friedt Professional Engineering Services, Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+
+static ZTEST_DMEM unsigned char __aligned(4) data[] = "abcdefgh";
+static struct k_pipe pipe = {
+	.buffer = data,
+	.size = sizeof(data) - 1 /* '\0' */,
+};
+
+static struct k_pipe bufferless;
+
+/*
+ *  r	 w
+ * |a|b|c|d|e|f|g|h|
+ * |0|1|2|3|4|5|6|7|
+ *
+ * r_avail = w - r = 3
+ * would read: a b c d
+ *
+ * w_avail = N - (w - r) = 5
+ * would overwrite: e f g h
+ */
+void test_pipe_avail_r_lt_w(void)
```

Comment:
no enforced or a common practice yet, but it would be great to have doxygen documentation for all tests (those that start with test_). We are going to use this for generating test plans and test documentation collateral. Basically a brief of what is being tested and short description of the test workflow. 

Thanks.

Reply:
@cfriedt could you please add the doxygen for those tests? Should be relatively simple, a matter of just documenting the purpose in the context of a Doxygen comment.

Reply:
Will do.